Rubberduck.VBEEditor/Events/VBENativeServices.cs

```diff
@@ -70,13 +70,22 @@ public static void UnhookEvents()
         public static void VbeEventCallback(IntPtr hWinEventHook, uint eventType, IntPtr hwnd, int idObject, int idChild,
             uint dwEventThread, uint dwmsEventTime)
         {
+            //This is an output window firehose, leave this here, but comment it out when done.
+            //if (idObject != (int)ObjId.Cursor) Debug.WriteLine("Hwnd: {0:X4} - EventType {1:X4}, idObject {2}, idChild {3}", (int)hwnd, eventType, idObject, idChild);
+
             if (hwnd != IntPtr.Zero && 
                 idObject == (int)ObjId.Caret && 
                 (eventType == (uint)WinEvent.ObjectLocationChange || eventType == (uint)WinEvent.ObjectCreate) &&
                 hwnd.ToWindowType() == WindowType.CodePane)
             {
                 OnSelectionChanged(hwnd);             
             }
+            else if (hwnd != IntPtr.Zero &&
+                (idObject == (int)ObjId.Caret && eventType == (uint)WinEvent.ObjectHide) &&
+                hwnd.ToWindowType() == WindowType.CodePane)
+            {
+                OnTypingCode(hwnd);
```

Comment:
may need a comment to indicate that this condition doesn't *only* fire when typing code.

Reply:
@WaynePhillipsEA this *sometimes* (inconsistently?) fires when merely navigating (by keyboard or mouse, makes no difference) to a line, which triggers the autocomplete service... any idea if it *can* be further filtered? I used the output window firehose to see what args were received when typing in the code pane, this seems as tight as it gets.